### DISCUSSION TITLE
FQDN torznab URL gets rewritten to `localhost`

### DISCUSSION THREAD

**Discussion by cartfisk** [cross-seed/cross-seed#535] [Q&A] [answered]
I have Prowlarr running behind an Apache2 reverse proxy and letsencrypt, i.e. `https://my.sub.domain.com/prowlarr`. When I use this URL scheme in the `torznab` block of `config.js`, the initial config check passes, but it eventually fails when searching. Somewhere along the line the sub-domain gets rewritten or defaults back to `http://localhost:9696`.

### config.js
```
...
    torznab: [
        "https://my.sub.domain.com/prowlarr/3/api?apikey=[REDACTED]",
        ...
    ],
...
```

### Log
`2023-10-29 02:34:43 error: failed to access http://localhost:9696/prowlarr/3/download?apikey=[REDACTED]&link=...`

I was able to work around this by using the machine's local IP address in the config, but it would be nice to be able to use a FQDN here instead of an IP if possible.

## Solution

I had my Prowlarr reverse proxy incorrectly configured. I was missing the following directives in my VirtualHost:

```
        ProxyPreserveHost On
        RequestHeader set "X-Forwarded-Proto" expr=%{REQUEST_SCHEME}
        RequestHeader set "X-Forwarded-SSL" expr=%{HTTPS}
```

**Comment by mmgoodnow** [comment]
FQDNs are supported. Your Prowlarr instance may not be set up correctly.

**Comment by mmgoodnow** [comment, marked as answer]
Maybe your reverse proxy isn't setting the right headers?

**Reply by cartfisk** [reply]
Could surely be. I'll mess around some and report back.

Thanks!

**Reply by mmgoodnow** [reply]
My guess is the `Host` header

**Reply by cartfisk** [reply]
I'll try adding the `ProxyPreserveHost` directive in apache and see if that works. I also noticed some `RequestHeader` directives mentioned in the Prowlarr docs that I didn't have so I'd imagine one of these will solve it.

I have a gigantic first time search running but I'll check back in when it's done.

Thanks for the tip!

**Reply by mmgoodnow** [reply]
ok yeah, I tried hitting my Prowlarr locally and by setting the `Host` header to "google.com", I was able to get it to spit out links starting with google.com

**Reply by cartfisk** [reply]
Awesome, I think in that case the preserve host directive will do the trick. Thank you kindly for digging into it with me.
